apple/Handlers/RNFlingHandler.m

```diff
@@ -158,17 +158,225 @@ - (RNGestureHandlerEventExtraData *)eventExtraData:(id)_recognizer
 
 #else
 
+@interface RNBetterSwipeGestureRecognizer : NSGestureRecognizer {
+  dispatch_block_t failFlingAction;
+  int maxDuration;
+  int minVelocity;
+  double defaultAlignmentCone;
+  double axialDeviationCosine;
+  double diagonalDeviationCosine;
+}
+
+@property (atomic, assign) RNGestureHandlerDirection direction;
+@property (atomic, assign) int numberOfTouchesRequired;
+
+- (id)initWithGestureHandler:(RNGestureHandler *)gestureHandler;
+
+@end
+
+@implementation RNBetterSwipeGestureRecognizer {
+  __weak RNGestureHandler *_gestureHandler;
+
+  NSPoint lastPosition;
+  double lastTime;
+}
+
+- (id)initWithGestureHandler:(RNGestureHandler *)gestureHandler
+{
+  if ((self = [super initWithTarget:self action:@selector(handleGesture:)])) {
+    _gestureHandler = gestureHandler;
+
+    maxDuration = 1.0;
+    minVelocity = 700;
+
+    defaultAlignmentCone = 30;
+    axialDeviationCosine = [self coneToDeviation:defaultAlignmentCone];
+    diagonalDeviationCosine = [self coneToDeviation:(90 - defaultAlignmentCone)];
+  }
+  return self;
+}
+
+- (void)handleGesture:(NSPanGestureRecognizer *)gestureRecognizer
+{
+  [_gestureHandler handleGesture:self];
+}
+
+- (void)mouseDown:(NSEvent *)event
+{
+  [super mouseDown:event];
+
+  lastPosition = [self locationInView:self.view];
+  lastTime = CACurrentMediaTime();
+
+  self.state = NSGestureRecognizerStateBegan;
+
+  __weak typeof(self) weakSelf = self;
+
+  failFlingAction = dispatch_block_create(0, ^{
+    __strong typeof(self) strongSelf = weakSelf;
+
+    if (strongSelf) {
+      strongSelf.state = NSGestureRecognizerStateFailed;
+    }
+  });
+
+  dispatch_after(
+      dispatch_time(DISPATCH_TIME_NOW, (int64_t)(maxDuration * NSEC_PER_SEC)),
+      dispatch_get_main_queue(),
+      failFlingAction);
+}
+
+- (void)mouseDragged:(NSEvent *)event
+{
+  [super mouseDragged:event];
+
+  self.state = NSGestureRecognizerStatePossible;
+
+  NSPoint currentPosition = [self locationInView:self.view];
+  double currentTime = CACurrentMediaTime();
+
+  NSPoint distance;
+  distance.x = currentPosition.x - lastPosition.x;
+  distance.y = lastPosition.y - currentPosition.y;
+
+  double timeDelta = currentTime - lastTime;
+
+  Vector *velocityVector = [Vector fromVelocityX:(distance.x / timeDelta) withVelocityY:(distance.y / timeDelta)];
```

Comment:
Isn't that very susceptible to sudden changes in velocity/direction?

Reply:
It may be. This is definitely not the best solution, but the other one that I see is to re-write [VelocityTracker](https://github.com/software-mansion/react-native-gesture-handler/blob/40b4e11efb4bef0417cf2e2f14df5bc17253f8da/src/web/tools/VelocityTracker.ts). Is this something that we want to do?

Reply:
I think we were talking about using the whole distance divided by the entire time of the gesture, would that be hard to implement? It should be a relatively good approximation on the time scale we're operating here.

If not, then using `delta` is better than calculating it ourselves. I see that you already changed it in https://github.com/software-mansion/react-native-gesture-handler/pull/3028/commits/79ab75c24bdd9a11049249dfddc3ac8aaca00c0f, could you then remove calculations for distance?

Reply:
Oops, I must have accidentally staged these changes as I remember that I wanted to ask you first about it 😅 

>I think we were talking about using the whole distance divided by the entire time of the gesture, would that be hard to implement?

Sounds like something that is possible, I'll check that 😅

Reply:
Okay, I've done it in [6a04fd6](https://github.com/software-mansion/react-native-gesture-handler/pull/3028/commits/6a04fd6ee2e35883a90323b4319310ca29712820), let me know if that is what you had in mind 😅

Reply:
As discussed in person, we will use calculating total distance instead of difference between `mouseDragged` events.